conda/fetch.py

```diff
@@ -305,6 +305,11 @@ def fetch_index(channel_urls, use_cache=False, unknown=False, index=None):
             futures = tuple(executor.submit(fetch_repodata, url, use_cache=use_cache,
                                             session=CondaSession()) for url in urls)
             repodatas = [(u, f.result()) for u, f in zip(urls, futures)]
+
+        except Exception as e:
```

Comment:
Actually, just above at line 298, someone has added this.

         # concurrent.futures is only available in Python >= 3.2 or if futures is installed
         # RuntimeError is thrown if number of threads are limited by OS

That's an awesome comment right after an `except`.  It's not always clear when an exception is expected.  That comment tells you exactly why we're catching both `ImportError` and `RuntimeError`.